business/istio_status.go

```diff
@@ -249,6 +259,20 @@ func (iss *IstioStatusService) getStatusOf(workloads []*models.Workload, cluster
 			continue
 		}
 
+		// Determine namespace to use
+		namespace := func() string {
+			if stat.Namespace != "" {
+				return stat.Namespace
+			}
+			return workload.Namespace
+		}()
+
+		// Create unique key for this component to prevent duplicates
+		componentKey := fmt.Sprintf("%s:%s:%s", cluster, namespace, workload.Name)
```

Comment:
```suggestion
		componentKey := strings.Join([]string{cluster, namespace, workload.name}, ":")
```

Slightly faster, but more importantly this is our convention for joining 3 or more things together with the same separator.